devnet-sdk/shell/env/kt_fetch.go

```diff
@@ -0,0 +1,88 @@
+package env
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/ethereum-optimism/optimism/kurtosis-devnet/pkg/deploy"
+	"github.com/ethereum-optimism/optimism/kurtosis-devnet/pkg/kurtosis/sources/artifact"
+)
+
+// EnclaveFS is an interface that both our mock and the real implementation satisfy
+type EnclaveFS interface {
+	GetArtifact(ctx context.Context, name string) (*artifact.Artifact, error)
+	Close() error
+}
+
+// enclaveFSWrapper wraps the artifact.EnclaveFS to implement our EnclaveFS interface
+type enclaveFSWrapper struct {
+	fs *artifact.EnclaveFS
+}
+
+func (w *enclaveFSWrapper) GetArtifact(ctx context.Context, name string) (*artifact.Artifact, error) {
+	return w.fs.GetArtifact(ctx, name)
+}
+
+func (w *enclaveFSWrapper) Close() error {
+	// The underlying EnclaveFS doesn't have a Close method, but we need it for our interface
+	return nil
+}
+
+// NewEnclaveFSFunc is the type for functions that create new enclave filesystems
+type NewEnclaveFSFunc func(ctx context.Context, enclave string) (EnclaveFS, error)
+
+// NewEnclaveFS is a variable that holds the function to create a new enclave filesystem
+// It can be replaced in tests
+var NewEnclaveFS NewEnclaveFSFunc = func(ctx context.Context, enclave string) (EnclaveFS, error) {
+	fs, err := artifact.NewEnclaveFS(ctx, enclave)
+	if err != nil {
+		return nil, err
+	}
+	return &enclaveFSWrapper{fs: fs}, nil
+}
+
+// parseKurtosisURL parses a Kurtosis URL of the form kt://enclave/artifact/file
+// If artifact is omitted, it defaults to "devnet"
+// If file is omitted, it defaults to "env.json"
+func parseKurtosisURL(u *url.URL) (enclave, artifactName, fileName string) {
+	enclave = u.Host
+	artifactName = deploy.DevnetEnvArtifactName
+	fileName = deploy.DevnetEnvArtifactPath
+
+	parts := strings.Split(strings.TrimPrefix(u.Path, "/"), "/")
+	if len(parts) > 0 && parts[0] != "" {
+		artifactName = parts[0]
+	}
+	if len(parts) > 1 {
+		fileName = parts[1]
+	}
```

Comment:
Have you missed the case of a trailing slash? (eg. "kt://enclave/artifact/")
I didn't see an associated test for it either.

```suggestion
	if len(parts) > 1 && parts[1] != "" {
		fileName = parts[1]
	}
```

Reply:
good catch, thanks !